// angular2-notification-bar/release/src/notification-bar.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NotificationBarService } from './notification-bar.service';
import { NotificationBarComponent, MESSAGES_CONFIG } from './notification-bar.component';
export var NotificationBarModule = (function () {
    function NotificationBarModule() {
    }
    NotificationBarModule.forRoot = function (config) {
        return {
            ngModule: NotificationBarModule,
            providers: [
                { provide: MESSAGES_CONFIG, useValue: config }
            ]
        };
    };
    NotificationBarModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [NotificationBarComponent],
                    providers: [NotificationBarService],
                    exports: [NotificationBarComponent]
                },] },
    ];
    /** @nocollapse */
    NotificationBarModule.ctorParameters = [];
    return NotificationBarModule;
}());
